tests/pytests/common.py

```diff
@@ -743,39 +743,47 @@ def check_index_info_empty(env, idx, fields, msg="after delete all and gc", dept
     expected_size = getInvertedIndexInitialSize_MB(env, fields, depth=depth+1)
     check_index_info(env, idx, exp_num_records=0, exp_inv_idx_size=expected_size, msg=msg, depth=depth+1)
 
-def downloadFiles(env, rdbs=None):
+
+def downloadFile(env, file_name, depth=0):
+    path = os.path.join(REDISEARCH_CACHE_DIR, file_name)
+    path_dir = os.path.dirname(path)
+    os.makedirs(path_dir, exist_ok=True)  # create dir if not exists
+    if not os.path.exists(path):
+        env.debugPrint(f"downloading {file_name}", force=True)
+        try:
+            subprocess.run(
+                [
+                    "wget",
+                    "--no-check-certificate",
+                    BASE_RDBS_URL + file_name,
+                    "-O",
+                    path,
+                    "-q",
+                ],
+                check=True,
+            )
+        except subprocess.CalledProcessError as e:
+            env.assertTrue(
+                False,
+                message=f"Failed to download {BASE_RDBS_URL + file_name}. Return code: {e.returncode}, output: {e.output}, stderr: {e.stderr}",
+                depth=depth + 1,
+            )
+            try:
+                os.remove(path)
+                env.debugPrint(f"Partially downloaded file {path}. Removing it.", force=True)
+            except OSError:
+                env.debugPrint(f"Failed to remove {path}", force=True)
+                pass
+            return False
+    if not os.path.exists(path):
+        return False
+    return True
```

Comment:
We can also just return `True` at this point no? No need to recheck if the file exists.
If I’m wrong, we can also just return the return value of `os.path.exists(path)`

Reply:
I also want to believe that handeling errors from wget covers all potential failures, but since we didn't fully identify the reason for the missing/corrupted files, i chose to keep this sanity check.
Ill add a log here